backend/src/main/java/com/bakdata/conquery/util/search/solr/SolrProcessor.java

```diff
@@ -0,0 +1,295 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import com.bakdata.conquery.apiv1.FilterTemplate;
+import com.bakdata.conquery.apiv1.LabelMap;
+import com.bakdata.conquery.apiv1.frontend.FrontendValue;
+import com.bakdata.conquery.io.storage.NamespaceStorage;
+import com.bakdata.conquery.models.datasets.Column;
+import com.bakdata.conquery.models.datasets.concepts.Searchable;
+import com.bakdata.conquery.models.datasets.concepts.filters.specific.SelectFilter;
+import com.bakdata.conquery.models.identifiable.ids.specific.FilterId;
+import com.bakdata.conquery.models.index.FrontendValueIndex;
+import com.bakdata.conquery.models.index.FrontendValueIndexKey;
+import com.bakdata.conquery.models.index.IndexCreationException;
+import com.bakdata.conquery.models.jobs.Job;
+import com.bakdata.conquery.models.jobs.UpdateFilterSearchJob;
+import com.bakdata.conquery.models.query.InternalFilterSearch;
+import com.bakdata.conquery.resources.api.ConceptsProcessor.AutoCompleteResult;
+import com.bakdata.conquery.util.progressreporter.ProgressReporter;
+import com.bakdata.conquery.util.search.Search;
+import com.bakdata.conquery.util.search.SearchProcessor;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.Sets;
+import io.dropwizard.util.Duration;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.time.StopWatch;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.util.ClientUtils;
+
+@Slf4j
+@RequiredArgsConstructor
+public class SolrProcessor implements SearchProcessor {
+
+	@NonNull
+	private final SolrClient solrClient;
+
+	private final Duration commitWithin;
+
+	private final int updateChunkSize;
+
+	private final String queryTemplate;
+
+	private final boolean combineEquallyNamedColumns;
+
+	private final Map<String, Search<FrontendValue>> searches = new ConcurrentHashMap<>();
+
+	@Override
+	public void clearSearch() {
+		try {
+			log.info("Clearing collection: {}", solrClient.getDefaultCollection());
+			solrClient.deleteByQuery("*:*");
+		}
+		catch (SolrServerException | IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override
+	public Job createUpdateFilterSearchJob(NamespaceStorage storage, Consumer<Set<Column>> columnsConsumer) {
+		return new UpdateFilterSearchJob(storage, this, columnsConsumer);
+	}
+
+	@Override
+	public void registerValues(Searchable<FrontendValue> searchable, Collection<String> values) {
+
+		SolrSearch search = (SolrSearch) getSearchFor(searchable);
+
+		search.registerValuesRaw(values);
+	}
+
+	@Override
+	public long getTotal(SelectFilter<?> filter) {
```

Comment:
schau mal bitte ob du das separat brauchst oder ob du das nicht in einem rutsch integriert abfragen könntest